### PR TITLE
Example Vessels junction with simplex and tracer

### PR DESCRIPTION
**Pull request by oguevremont**
This should be a complete example showing on how to generate and use a simplex grid, while also being an example of the tracer physics. Please view this page (https://github.com/lethe-cfd/lethe/wiki/Example-16-:-Vessels-Junction-with-Tracer-Physics-on-a-Simplex-Mesh) accompanying it. Suggestions/comments are welcome.

### CODE REVIEW

File: examples/incompressible_flow/3d_vessels/vessels.prm [hunk 1]
```diff
@@ -0,0 +1,140 @@
+# Listing of Parameters
+# ---------------------
+
+
```
Comment (blaisb) [inline examples/incompressible_flow/3d_vessels/vessels.prm:4]:
```suggestion
```

File: examples/incompressible_flow/3d_vessels/vessels.prm [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+end
+
+#---------------------------------------------------
+# FEM
+#---------------------------------------------------
+subsection FEM
+    set velocity order            = 1
+    set pressure order            = 1
+    set tracer order              = 1
+    set qmapping all              = false
```
Comment (blaisb) [inline examples/incompressible_flow/3d_vessels/vessels.prm:24]: qmapping is set to false by default, do you really want to keep that option visible here and talk about it or not?

File: examples/incompressible_flow/3d_vessels/vessels.prm [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    set file name            = vessels.msh
+    set initial refinement   = 0
+    set simplex              = true
+end
+
+# --------------------------------------------------
+# Boundary Conditions
+#---------------------------------------------------
+subsection boundary conditions
+  set number                  = 4
```
Comment (blaisb) [inline examples/incompressible_flow/3d_vessels/vessels.prm:60]: Why number = 4 if you only set 3 three boundary conditions?
You need to have number = number of boundary conditions you actually set.

File: examples/incompressible_flow/3d_vessels/vessels.prm [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            set Function expression = 4
+        end
+    end
+    subsection bc 2
+        set id = 3
+        set type              = noslip
+    end
+end
+subsection boundary conditions tracer
+  set number                  = 3
```
Comment (blaisb) [inline examples/incompressible_flow/3d_vessels/vessels.prm:94]:
```suggestion
  set number                  = 2
```

File: examples/incompressible_flow/3d_vessels/vessels.prm [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+              set Function expression =  1 
+      end
+    end
+end
+
+# --------------------------------------------------
+# Non-Linear Solver Control
+#---------------------------------------------------
+subsection non-linear solver
+  set tolerance               = 1e-4
```
Comment (blaisb) [inline examples/incompressible_flow/3d_vessels/vessels.prm:115]:
```suggestion
  set tolerance               = 1e-6
```

File: examples/incompressible_flow/3d_vessels/vessels.prm [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+  set relative residual         = 1e-3
+  set minimum residual          = 1e-11
+  set amg preconditioner ilu fill                  = 0
+  set amg preconditioner ilu absolute tolerance    = 1e-10
+  set amg preconditioner ilu relative tolerance    = 1.00
+  set amg aggregation threshold = 1e-14  # Aggregation
+  set amg n cycles              = 1      # Number of AMG cycles
+  set amg w cycles              = false  # W cycles, otherwise V cycles
+  set amg smoother sweeps       = 2      # Sweeps
+  set amg smoother overlap      = 1      # Overlap
```
Comment (blaisb) [inline examples/incompressible_flow/3d_vessels/vessels.prm:136]:
```suggestion
```

**Review by blaisb — COMMENTED** [review]
Minor comments to address, but otherwise looking good. If you can address that today I'll merge later today

File: examples/incompressible_flow/3d_vessels/vessels.prm (hunk 2 above)
Comment (oguevremont) [reply examples/incompressible_flow/3d_vessels/vessels.prm]: I wouldn't want to talk about it

**Pull request merged**